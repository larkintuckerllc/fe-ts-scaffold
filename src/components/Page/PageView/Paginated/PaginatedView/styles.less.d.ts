export const previous: string;
export const next: string;
