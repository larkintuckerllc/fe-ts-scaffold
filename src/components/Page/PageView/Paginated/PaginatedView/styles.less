#previous {
  display: block;
}
#next {
  display: block;
}