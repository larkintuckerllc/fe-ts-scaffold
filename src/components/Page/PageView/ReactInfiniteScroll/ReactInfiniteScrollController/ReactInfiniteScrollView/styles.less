.rootItem {
  height: 300px;
  border: 1px solid black;
}