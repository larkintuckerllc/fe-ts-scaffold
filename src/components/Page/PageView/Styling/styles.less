@import '~antd/lib/style/themes/default.less';

#root {
  background-color: white;
}
#rootContainer {
  display: flex;
}
.rootContainerItem {
  color: @primary-color;
  background-color: @normal-color;
}