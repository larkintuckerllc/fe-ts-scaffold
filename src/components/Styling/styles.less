#root {
  background-color: white;
}
.rootItem {
  color: @primary-color;
}