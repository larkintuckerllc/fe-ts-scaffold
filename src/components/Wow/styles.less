.wow {
  color: @primary-color;
  background-color: red;
}