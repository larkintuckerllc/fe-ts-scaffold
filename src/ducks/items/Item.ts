import { Record } from 'immutable';

const itemDefault = {
  id: 0,
  name: 'name',
};

export interface ItemJS {
  id: number;
  name: string;
}

export default class Item extends Record(itemDefault) {
  constructor(params: ItemJS) {
    super(params);
  }
  public get<T extends keyof ItemJS>(value: T): ItemJS[T] {
    return super.get(value, null);
  }
}
