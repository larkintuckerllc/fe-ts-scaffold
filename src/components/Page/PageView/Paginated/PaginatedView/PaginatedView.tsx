import Item from 'DUCKS/items/Item';
import { List } from 'immutable';
import React, { Component } from 'react';
import Items from './Items';

interface PaginatedViewProps {
  currentPage: number;
  error: boolean;
  requested: boolean;
  lastPage: number;
  items: List<Item>;
  fetchItems(page: number): void;
}

export default class AsyncView extends Component<PaginatedViewProps> {
  public componentDidMount() {
    const { currentPage, fetchItems } = this.props;
    fetchItems(currentPage);
  }
  public render() {
    const { currentPage, error, lastPage, requested, items } = this.props;
    if (requested) {
      return <div>Requested</div>;
    } else if (error) {
      return <div>Error</div>;
    } else if (items.size === 0) {
      return <div>No Items</div>;
    }
    return (
      <div>
        <h2>Paginated</h2>
        <Items items={items.toJS()} />
        <div>CurrentPage: {currentPage.toString()}</div>
        <div>LastPage: {lastPage.toString()}</div>
        {currentPage !== 0 && (
          <div>
            <button onClick={this.handlePreviousPage}>Previous Page</button>
          </div>
        )}
        {currentPage !== lastPage && (
          <div>
            <button onClick={this.handleNextPage}>Next Page</button>
          </div>
        )}
      </div>
    );
  }

  private handleNextPage = () => {
    const { currentPage, fetchItems } = this.props;
    fetchItems(currentPage + 1);
  };

  private handlePreviousPage = () => {
    const { currentPage, fetchItems } = this.props;
    fetchItems(currentPage - 1);
  };
}
