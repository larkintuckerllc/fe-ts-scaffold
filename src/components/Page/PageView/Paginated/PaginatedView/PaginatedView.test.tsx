import itemsTestData from 'APIS/items/items.testdata';
import Item, { ItemFactory, ItemRecord } from 'DUCKS/items/Item';
import Enzyme, { shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import { List } from 'immutable';
import React from 'react';
import PaginatedView from './PaginatedView';
import styles from './styles.less';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

const itemsTestDataPaged = itemsTestData.slice(0, 2);
const reducer = (accumulator: List<ItemRecord>, jsonItem: Item) =>
  accumulator.push(ItemFactory(jsonItem));
const itemsSample = itemsTestDataPaged.reduce(reducer, List<ItemRecord>([])) as List<ItemRecord>;
const getDefaultProps = () => ({
  currentPage: 0,
  error: false,
  fetchItems: jest.fn(),
  items: itemsSample,
  lastPage: 1,
  requested: false,
});

describe('PaginatedView component', () => {
  it('shallow renders without crashing', () => {
    const defaultProps = getDefaultProps();
    shallow(<PaginatedView {...defaultProps} />);
  });
  it('renders differently with requested', () => {
    const { requested, ...defaultProps } = getDefaultProps();
    const wrapper = shallow(<PaginatedView {...defaultProps} requested={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders differently with not requested and error', () => {
    const { error, ...defaultProps } = getDefaultProps();
    const wrapper = shallow(<PaginatedView {...defaultProps} error={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders differently with not requested not error and 0 todos', () => {
    const { items, ...defaultProps } = getDefaultProps();
    const emptyItems = List<ItemRecord>([]);
    const wrapper = shallow(<PaginatedView {...defaultProps} items={emptyItems} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders differently with currentPage not 0', () => {
    const { currentPage, ...defaultProps } = getDefaultProps();
    const testCurrentPage = 1;
    const wrapper = shallow(<PaginatedView {...defaultProps} currentPage={testCurrentPage} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders differently with currentPage not 0 and lastPage', () => {
    const { currentPage, lastPage, ...defaultProps } = getDefaultProps();
    const testCurrentPage = 1;
    const testLastPage = 1;
    const wrapper = shallow(
      <PaginatedView {...defaultProps} currentPage={testCurrentPage} lastPage={testLastPage} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls fetchItems on mount', () => {
    const { currentPage, fetchItems, ...defaultProps } = getDefaultProps();
    shallow(<PaginatedView {...defaultProps} currentPage={currentPage} fetchItems={fetchItems} />);
    expect(fetchItems.mock.calls.length).toBe(1);
    expect(fetchItems.mock.calls[0][0]).toBe(currentPage);
  });

  it('calls fetchItems on next click', () => {
    const { fetchItems, lastPage, ...defaultProps } = getDefaultProps();
    const testLastPage = 1;
    const wrapper = shallow(
      <PaginatedView {...defaultProps} fetchItems={fetchItems} lastPage={testLastPage} />
    );
    const testNext = wrapper.find(`#${styles.next}`);
    testNext.simulate('click');
    const callsLength = 2;
    const nextPage = 1;
    expect(fetchItems.mock.calls).toHaveLength(callsLength);
    expect(fetchItems.mock.calls[1][0]).toBe(nextPage);
  });

  it('calls fetchItems on previous click', () => {
    const { currentPage, fetchItems, lastPage, ...defaultProps } = getDefaultProps();
    const testCurrentPage = 1;
    const testLastPage = 1;
    const wrapper = shallow(
      <PaginatedView
        {...defaultProps}
        currentPage={testCurrentPage}
        fetchItems={fetchItems}
        lastPage={testLastPage}
      />
    );
    const testPrevious = wrapper.find(`#${styles.previous}`);
    testPrevious.simulate('click');
    const callsLength = 2;
    const previousPage = 0;
    expect(fetchItems.mock.calls).toHaveLength(callsLength);
    expect(fetchItems.mock.calls[1][0]).toBe(previousPage);
  });
});
