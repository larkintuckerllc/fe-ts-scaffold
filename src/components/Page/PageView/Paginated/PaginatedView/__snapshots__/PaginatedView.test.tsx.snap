// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedView component renders differently with currentPage not 0 1`] = `
<div>
  <h2>
    Paginated
  </h2>
  <Items
    items={
      Array [
        Object {
          "id": 0,
          "name": "name",
        },
      ]
    }
  />
  <div>
    CurrentPage: 
    1
  </div>
  <div>
    LastPage: 
    0
  </div>
  <div>
    <button
      id="previous"
      onClick={[Function]}
    >
      Previous Page
    </button>
  </div>
  <div>
    <button
      id="next"
      onClick={[Function]}
    >
      Next Page
    </button>
  </div>
</div>
`;

exports[`PaginatedView component renders differently with currentPage not 0 and lastPage 1`] = `
<div>
  <h2>
    Paginated
  </h2>
  <Items
    items={
      Array [
        Object {
          "id": 0,
          "name": "name",
        },
      ]
    }
  />
  <div>
    CurrentPage: 
    1
  </div>
  <div>
    LastPage: 
    1
  </div>
  <div>
    <button
      id="previous"
      onClick={[Function]}
    >
      Previous Page
    </button>
  </div>
</div>
`;

exports[`PaginatedView component renders differently with not requested and error 1`] = `
<div>
  Error
</div>
`;

exports[`PaginatedView component renders differently with not requested not error and 0 todos 1`] = `
<div>
  No Items
</div>
`;

exports[`PaginatedView component renders differently with requested 1`] = `
<div>
  Requested
</div>
`;
